runtime/src/commonJvmAndroid/kotlin/pbandk/MessageExtensionsJvm.kt

```diff
@@ -33,39 +36,26 @@ public fun <T : Message> Message.Companion<T>.decodeFromByteBuffer(buffer: ByteB
 /**
  * Read the next message of type T, in the [stream] of size-delimited Messages, such as is written by the Java method
  * AbstractMessageLite.writeDelimitedTo(OutputStream) and read by Message.parseDelimitedFrom(InputStream).
- * A null is returned when all messages have been read, and the stream is closed.
+ * A null is returned when all messages have been read. The caller must close the stream.
  */
-public fun <T : Message> Message.Companion<T>.parseDelimitedFrom(stream: InputStream): T? {
+public fun <T : Message> Message.Companion<T>.decodeDelimitedFromStream(stream: InputStream): T? {
     val length = readVlen(stream)
     if (length < 0) {
-        stream.close()
         return null
```

Comment:
Sorry I didn't catch this on the first pass: This will incorrectly return `null` even if the EOF happens in the middle of decoding the varint. If the EOF happens in the middle of varint decoding, then the code should `throw InvalidProtocolBufferException.truncatedMessage()` (similar to what `KotlinBinaryWireDecoder.readRawBytes()` does). This code should only return `null` if the EOF happens before any varint decoding starts.

My suggestion would be:
1. Modify `readerRawVarint64()` to take the first byte as an argument, along with the lambda for reading additional bytes.
2. Check for the EOF condition in this method before calling `readVlen(stream)` by first calling `stream.read()` to read the next byte and see if it's `-1`.
3. If the first byte wasn't -1, then pass it into the `readerRawVarint64()` function, along with the lambda that calls `input.read()` to get additional bytes.